### PR TITLE
[8.0]-[BR1030]-[ADD/IMP][Reference field improvements]

### PR DESCRIPTION
**Pull request by serpentcs-dev1**
Issue:- #184 

- Improved string Reference to "WBS"

- Improved name_get method

- Added origin field with text type

@elicoidal @victormartinelicocorp

### CODE REVIEW

File: business_requirement/models/business.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def _get_default_company(self):
         states={'draft': [('readonly', False)]}
     )
     ref = fields.Char(
-        'Reference',
+        'WBS',
         required=False,
         readonly=True,
```
Comment (elicoidal) [inline business_requirement/models/business.py:45]: Any reason why r/o?

**Review by elicoidal — APPROVED** [review]
LGTM 
@victormartinelicocorp @seb-elico

File: business_requirement/models/business.py [hunk 2]
```diff
@@ -269,7 +270,8 @@ def name_get(self):
         result = []
         for br in self:
             if br.ref:
-                formatted_name = u'[{}] {}'.format(br.ref, br.description)
+                formatted_name = u'[{}][{}] {}'.format(br.ref, br.name,
```
Comment (martintorresvm) [inline business_requirement/models/business.py:273]: @darshan-serpent What is the point of have twice the br.name? Perhaps you forgot add other field?

**Review by martintorresvm — CHANGES_REQUESTED** [review]

File: business_requirement/models/business.py (hunk 1 above)
Comment (serpentcs-dev1) [reply business_requirement/models/business.py:45]: Hmm.. I haven't changed it was already there.

But yes, I think read-only should not be there, should I remove?

File: business_requirement/models/business.py (hunk 2 above)
Comment (serpentcs-dev1) [reply business_requirement/models/business.py:273]: Yes you are right. I think It should be like, right? 

`formatted_name = u'[{}] {}'.format(br.ref, br.name, br.description)`

Will do the needful, thanks!

File: business_requirement/models/business.py (hunk 1 above)
Comment (elicoidal) [reply business_requirement/models/business.py:45]: I am not sure as there are some ACL based on status. I dont know if it has an impact so dont modify it here.

File: business_requirement/models/business.py (hunk 2 above)
Comment (elicoidal) [reply business_requirement/models/business.py:273]: not sure here but if the requirement is fulfilled, this is fine by me.

I want: [WBS][name] Description or [name] Description (if WBS empty)

Comment (martintorresvm) [reply business_requirement/models/business.py:273]: :+1:

**Comment by serpentcs-dev1** [comment]
@elicoidal @victormartinelicocorp Tested on runbot, works as expected.

The test is made with 2 BRs, one with WBS and other without it.

without_wbs
with_wbs

**Review by elicoidal — APPROVED** [review]
Perfect!

**Comment by elicoidal** [comment]
@victormartinelicocorp please review again

**Review by martintorresvm — APPROVED** [review]

**Pull request merged**